lib/hasPermissions.ts

```diff
@@ -203,16 +207,57 @@ export class HasPermissionsDirectiveVisitor<
     const hasPermissionsValidateFunction: ValidateFunction<TContext> = (
       value: unknown,
       _: unknown,
-      __: unknown,
+      container: GraphQLArgument | GraphQLInputObjectType | GraphQLObjectType,
       context: TContext,
       resolverInfo: Record<string, unknown>,
       resolverSource: unknown,
       resolverArgs: Record<string, unknown>,
+      path?: Array<string>,
     ): unknown => {
+      if (
+        (location === DirectiveLocation.INPUT_FIELD_DEFINITION ||
+          location === DirectiveLocation.INPUT_OBJECT) &&
+        container instanceof GraphQLInputObjectType &&
+        path &&
+        value instanceof Array
+      ) {
+        const verify = [];
+        for (let i = 0; i < value.length; i += 1) {
+          if (
+            value[i] !==
+            container.getFields()[path[path.length - 1]].defaultValue[i]
+          ) {
+            verify.push(false);
+          } else {
+            verify.push(true);
+          }
+        }
+        if (!verify.includes(false)) {
```

Comment:
this verify array looks strange. What are you trying to achieve? On top of that doing repeated searches on array (using the includes) is not very good. We should probably use a better approach.

Reply:
I agree, and changed this approach to another, where I use just one variable to check this.